src/porepy/fracs/fractures.py

```diff
@@ -2152,6 +2268,47 @@ def to_gmsh(self, file_name, **kwargs):
                             intersection_points=intersection_points,
                             mesh_size_bound=mesh_size_bound,
                             mesh_size=mesh_size, tolerance=gmsh_tolerance,
-                            edges_2_frac=self.decomposition['line_in_frac'])
+                            edges_2_frac=self.decomposition['line_in_frac'],
+                            fracture_tags=frac_tags)
 
         writer.write_geo(file_name)
+
+
+    def _on_boundary(self, edges, edge_tags):
```

Comment:
Perhaps rather on_domain_boundary? If you search for boundary in this module, you will get all kind of hits, so specification would not hurt.